client/src/store/index.ts

```diff
@@ -499,6 +505,30 @@ const {
     setShowCrosshairs(state, show) {
       state.showCrosshairs = show;
     },
+    switchReviewMode(state, mode){
+      state.reviewMode = mode || false;
+      if(mode){
+        const myRole = state.currentTaskOverview.my_project_role;
+        let scanStatesForMyReview = []
+        if(myRole == 'tier_2_reviewer'){
+          scanStatesForMyReview = ['needs tier 2 review']
+        } else if(myRole == 'tier_1_reviewer'){
+          scanStatesForMyReview = ['unreviewed']
+        }
+        const scanIdsForMyReview = Object.entries(state.currentTaskOverview.scan_states).filter(
```

Comment:
That depends. If we store the scan state on the Scan object, when do we want to update it? 

Right now, scan states are stored on this `currentTaskOverview` object, which is only updated when we select a Project. That means as a user goes through scans in review mode and makes reviews, the list stays the same until they switch project contexts or refresh the page. So in review mode, they can go back and forth between Scans even after they make reviews on them. The Scans don't disappear from their "needs my review" list immediately after they submit a `ScanDecision`. 

If we store the scan state on the Scan object, do we want to update that immediately as they make `ScanDecision`s? That would mean that those scans will disappear from their "needs my review" list immediately, and they couldn't go back to that scan without going to "All Scans" mode. 

OR, we could store the state on the Scan object without updating it upon creation of `ScanDecision`s. I'm not for or against any of these, but we should just keep in mind that the way we store Scan states will affect the user experience in Review Mode. What do you think?

Reply:
That's a compelling argument, I think it's better to be able to go back to previously reviewed scans in case mistakes were made. Deliberately keeping `Scan`s out of date to provide this behavior sounds more complicated to me than just using `currentTaskOverview` as we do now.